File: daq/slc/dqm/src/TimedGraph1I.cc
#include "daq/slc/dqm/TimedGraph1I.h"

#include "daq/slc/dqm/IntArray.h"

using namespace Belle2;

TimedGraph1I::TimedGraph1I() throw() : TimedGraph1()
{
  _data = new IntArray(0);
}

TimedGraph1I::TimedGraph1I(const TimedGraph1I& h) throw() : TimedGraph1(h)
{
  _data = new IntArray(h._data->size());
}

TimedGraph1I::TimedGraph1I(const std::string& name, const std::string& title,
                           int nbinx, double xmin, double xmax)
throw() : TimedGraph1(name, title, nbinx, xmin, xmax)
{
  _data = new IntArray(nbinx);
  _time_v.resize(nbinx);
}

TimedGraph1I::~TimedGraph1I() throw() {}

std::string TimedGraph1I::getDataType() const throw()
{
  return std::string("TGI");
}

File: daq/slc/extra/javalib/b2daq/src/b2daq/dqm/core/TimedGraph1I.java
package b2daq.dqm.core;

public class TimedGraph1I extends TimedGraph1 {

	public TimedGraph1I() {
		super();
		data = new IntArray();
	}

	public TimedGraph1I(String name, String title, int nbins, double xmin, double xmax) {
		super(name, title, nbins, xmin, xmax);
		data = new IntArray(nbins);
	}

	public String getDataType() {
		return "TGI";
	}

}
